coreneuron/io/reports/report_event.cpp

```diff
@@ -22,25 +23,44 @@ namespace coreneuron {
 ReportEvent::ReportEvent(double dt,
                          double tstart,
                          const VarsToReport& filtered_gids,
-                         const char* name)
+                         const char* name,
+                         double report_dt)
     : dt(dt)
     , tstart(tstart)
-    , report_path(name) {
+    , report_path(name)
+    , report_dt(report_dt) {
     VarsToReport::iterator it;
     nrn_assert(filtered_gids.size());
     step = tstart / dt;
+    reporting_period = static_cast<int>(report_dt / dt);
     gids_to_report.reserve(filtered_gids.size());
     for (const auto& gid: filtered_gids) {
         gids_to_report.push_back(gid.first);
     }
     std::sort(gids_to_report.begin(), gids_to_report.end());
 }
 
+void ReportEvent::summation_ALU(NrnThread* nt) {
+    if (static_cast<int>(step) % reporting_period == 0) {
+        auto* alu_mapping = static_cast<ALUMapping*>(nt->alu_);
+        auto& alu = alu_mapping->report_ALU_[report_path];
+        double sum = 0.0;
+        for (const auto& kv: alu.currents_) {
```

Comment:
nitpick : add here comment about why we need to do this / implementation details
(Q. do we also sum values from all compartments in a section into a single value?)